service/s3/presign_post_test.go

```diff
@@ -66,6 +68,23 @@ func TestPresignPutObject(t *testing.T) {
 			},
 			expectedURL: "https://mfzwi23gnjvgw.mrap.accesspoint.s3-global.amazonaws.com",
 		},
+		"use path style bucket hosting pattern": {
```

Comment:
Can we also add tests for:
* bucket and key are the same value
* custom BaseEndpoint has an arbitrary path segment (e.g. `locals3.com/v1`)

Reply:
Added tests, thanks for the suggestion